src/config/file/v1.rs

```diff
@@ -190,10 +226,41 @@ impl TryFrom<Integration> for remote::Integration {
                     .team(team)
                     .title(title)
                     .protocol(protocol)
+                    .import_branches(import_branches)
+                    .import_tags(import_tags)
+                    .watched_branches(watched_branches)
                     .build()
-                    .wrap_ok()
+            }
+        };
+
+        if integration.watched_branches().is_empty()
+            && integration.import_branches() == &remote::BranchImportStrategy::Enabled
+        {
+            let references =
+                futures::executor::block_on(integration.references()).unwrap_or_default();
```

Comment:
This is another case for "we should just stop implementing `TryFrom`, as it's not working for us".